pkg/tests/end_to_end_tests/create_test.go

```diff
@@ -577,6 +579,119 @@ func TestInsertAbort(t *testing.T) {
 	})
 }
 
+// Trigger the backround job and eviction simultaneously with a randomized overlap
+func TestEvictionVInsert(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	// Interate until we encounter all acceptable outcomes
+	// (S1, S1 + S2 (ressurection), S1 + S3 (new series id allocation).
+	expectedSeriesIds := map[int64]bool{}
+	for len(expectedSeriesIds) < 3 {
+		withDB(t, *testDatabase, func(dbOwner *pgxpool.Pool, t testing.TB) {
+			db := testhelpers.PgxPoolWithRole(t, *testDatabase, "prom_admin")
+			defer db.Close()
+
+			ingestor, err := ingstr.NewPgxIngestorForTests(pgxconn.NewPgxConn(db), nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer ingestor.Close()
+
+			// Shrink epoch duration
+			_, err = db.Exec(context.Background(), "SELECT _prom_catalog.set_default_value('epoch_duration', (INTERVAL '1 second')::TEXT)")
+			require.NoError(t, err)
+
+			farFuture := int64((time.Hour / time.Millisecond * 24 * 365 * 3000))
+			insertSamples := func(t0 int64, labelValue string, ignoreError bool) {
+				insertIters := int64(50)
+				samples := make([]prompb.TimeSeries, 0, insertIters)
+				for i := int64(0); i < insertIters; i++ {
+					samples = append(samples, prompb.TimeSeries{
+						Labels: []prompb.Label{
+							{Name: model.MetricNameLabelName, Value: "test"},
+							{Name: "test", Value: labelValue},
+						},
+						Samples: []prompb.Sample{
+							{Timestamp: t0 + i, Value: 0.1},
+						},
+					})
+				}
+				_, _, err := ingestor.IngestMetrics(context.Background(), newWriteRequestWithTs(copyMetrics(samples)))
+				if !ignoreError {
+					require.NoError(t, err)
+				}
+			}
+
+			// Ingest some metrics
+			insertSamples(farFuture, "test0", false) // this should survive
+			insertSamples(0, "test1", false)         // this should expire
+			err = ingestor.CompleteMetricCreation(context.Background())
+			require.NoError(t, err)
+
+			// An epoch duration delay
+			time.Sleep(1 * time.Second)
+
+			var wg sync.WaitGroup
+			wg.Add(3)
+
+			bgDelay := time.Duration(rand.Intn(90)) * time.Millisecond
```

Comment:
Can you please add comments for delays. I believe I do understand what they mean but still better to be explicit

Reply:
Done